src/hera/workflows/script.py

```diff
@@ -233,6 +205,8 @@ def _build_script(self) -> _ModelScriptTemplate:
             if not env:
                 env = []
             env.append(EnvVar(name="hera__script_annotations", value=""))
+            if hera_outputs_used:
```

Comment:
Seeing as we're doing `isinstance(self.constructor, RunnerScriptConstructor)`, I think it makes sense for this code (L201 to L209) to be in the RunnerScriptConstructor's `transform_script_template_post_build`? (i.e. we need to write and override the function as it doesn't exist in the subclass atm)